docs/developers/architecture/app_model.md

```diff
@@ -2,66 +2,195 @@
 
 # napari's application model
 
-```{important}
-**This is not a part of the public napari interface!**
```

Comment:
No, but it is now inside the 'architecture' section which only includes docs aimed at devs, so we can infer that it is not part of the interface. I guess it's good to spell it out.

Reply:
But as I understand, there is intention to make it public:
https://github.com/napari/napari/issues/6054

Without making get_app public, the actions cannot be simply called from plugins.

Reply:
If we do make it public we can amend this message.

I think the intention is more to allow users to register actions at runtime. Depending on the contribution type, the user may already have a public API func to access plugin actions- e.g., there is the `open_sample` func for sample contributions, and I think there may be a way for the user to get reader/writers too?

Reply:
I think that option to call actions is as important, as runtime registration

Reply:
You think users should call a command in `app` via the command ID string instead of using existing public API like `open_sample`? Why?
Or do you just mean you want users to have the option of calling a command via string?

Reply:
Because we provide public API only for opening sample and adding plugin widget. And plugins could provide many more contributions. 

And there is an area for people who would like to provide custom controllers (either widget or hardware)

Reply:
Sure but these are not the current comtribution types? We could create public API for new contribution types. Have we already decided how these new contributions would look?

I'm not opposed to execute via app model but there's probably a few decisions to be made here and this way may not be the best?

Reply:
Yeah I don't think this is going to get hashed out in this PR lol.

Let's update the message to: This is not currently a part of the public napari API, but we are considering ways to expose action execution directly. 

Once we actually decide what the interface for that is, we will amend.